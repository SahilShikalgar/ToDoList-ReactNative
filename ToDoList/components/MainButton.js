import React from 'react'
import { TouchableOpacity, TouchableNativeFeedback, Platform, StyleSheet, View, Text } from 'react-native';
import Colors from '../constants/Colors';

export default function MainButton(props) {
    let ButtonComponent = TouchableOpacity;

    if (Platform.Version >= 21) {
        ButtonComponent = TouchableNativeFeedback;
    }

    return (
        <View style={styles.buttonContainer}>
            <ButtonComponent activeOpacity={0.6} onPress={props.onPress}>
                <View style={styles.button}>
                    <Text style={styles.buttonText}>{props.children}</Text>
                </View>
            </ButtonComponent>
        </View>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        borderRadius: 25
    },
    button: {
        backgroundColor: Colors.primary,
        paddingVertical: 15,
        paddingHorizontal: 45,
        borderRadius: 25
    },
    buttonText: {
        color: 'white',
        fontFamily: 'lato',
        fontSize: 24,
        textAlign: 'center'
    }
});
