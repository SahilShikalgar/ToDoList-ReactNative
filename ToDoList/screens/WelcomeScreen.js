import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import MainButton from '../components/MainButton';
 
const WelcomeScreen = (props) => {
    console.log(props);
    const goToMainScreen = () => {
        props.navigation.navigate({routeName: 'Main'});
    }

    return (
        <View style={styles.screen}>
            <Text style={styles.title}>Welcome</Text>
            <View style={styles.buttonContainer}>
                <MainButton onPress={goToMainScreen}>NEXT</MainButton>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        padding: 10,
        alignItems: 'center',
        justifyContent: 'center'
    },
    title: {
        fontFamily: 'lato',
        fontWeight: '100',
        fontSize: 48
    },
    buttonContainer: {
        marginTop: 300,
        width: 150,
        borderColor: 'black',
        borderRadius: 25
    }
});

export default WelcomeScreen;
