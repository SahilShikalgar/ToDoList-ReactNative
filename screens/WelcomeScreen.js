import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import MainButton from '../components/MainButton';
 
const WelcomeScreen = (props) => {
    const goToMainScreen = () => {
        props.navigation.navigate({routeName: 'Main'});
    }

    return (
        <View style={styles.screen}>
            <Text style={styles.title}>Welcome</Text>
            <View style={styles.buttonContainer}>
                <MainButton onPress={goToMainScreen} style={styles.button}>
                    NEXT
                </MainButton>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        padding: 10,
        alignItems: 'center',
        justifyContent: 'center'
    },
    title: {
        fontFamily: 'lato',
        fontWeight: '100',
        fontSize: 48
    },
    buttonContainer: {
        marginTop: 300,
        borderColor: 'black',
        borderRadius: 25,
        flexDirection: "row"
    },
    button: {
        alignSelf: "center",
        width: "50%"
    }
});

export default WelcomeScreen;
