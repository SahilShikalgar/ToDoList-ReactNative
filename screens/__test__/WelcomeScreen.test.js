import React from 'react';
import { shallow } from 'enzyme';
import { WelcomeScreen } from '../WelcomeScreen';
import MainButton from '../../components/MainButton'

describe('WelcomeScreen', () => {
    const navigation = {
        navigate: jest.fn()
    }
    const component = shallow (
        <WelcomeScreen navigation={navigation} />
    );

    it('should render properly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Main Button Exists', () => {
        const mainButton = component.find(MainButton)
        expect(mainButton).toHaveLength(1)
    });

    it('Touchable Opacity Exists', () => {
        component.find(MainButton).prop('onPress')()
        expect(navigation.navigate).toHaveBeenCalled()
    })
});
