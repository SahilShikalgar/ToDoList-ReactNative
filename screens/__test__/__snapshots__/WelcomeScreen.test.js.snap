// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should rendere properly 1`] = `ShallowWrapper {}`;
